core/src/main/java/com/netease/arctic/catalog/BaseArcticCatalog.java

```diff
@@ -425,6 +427,15 @@ public Transaction newCreateTableTransaction() {
     }
 
     protected void doCreateCheck() {
+      if (primaryKeySpec.primaryKeyExisted()) {
+        Set<String> pkSet = new HashSet<>(primaryKeySpec.fieldNames());
+        schema.columns().forEach(nf -> {
+          if (pkSet.contains(nf.name()) && nf.isOptional()) {
+            throw new IllegalStateException("please check your schema, the primary key nested field must be required " +
```

Comment:
`IllegalArgumentException` would be more suitable？

Reply:
Has been modified.